ext/nnstreamer/tensor_filter/tensor_filter_flare.cc

```diff
@@ -0,0 +1,377 @@
+  result = context->setSFlareOptions (config);
+  if (result != SFlareApi::ErrorCode::SFLARE_SUCCESS) {
+    throw std::runtime_error ("Failed to set flare options");
+  }
+
+  result = context->loadSFlareLLMModel (backend, use_gpu);
+  if (result != SFlareApi::ErrorCode::SFLARE_SUCCESS) {
+    throw std::runtime_error ("Failed to load flare model");
+  }
+}
+
+/**
+ * @brief Configure flare instance
+ */
+void
+TensorFilterFlare::configure_instance (const GstTensorFilterProperties *prop)
+{
+  if (!prop->invoke_dynamic) {
+    throw std::invalid_argument (
+        "Flare only supports invoke-dynamic mode. Set `invoke-dynamic=true`");
+  }
+
+  try {
+    parseCustomProperties (prop);
+  } catch (const std::invalid_argument &e) {
+    throw std::invalid_argument ("Failed to parse \"custom\" prop:"
+                                 + std::string (e.what ()) + "\n\tReference: " + __FILE__);
+  }
+  try {
+    initializeFlareContext ();
+  } catch (const std::exception &e) {
+    throw std::runtime_error (
+        "Failed to initialize flare context: " + std::string (e.what ()));
+  }
+
+  std::cout << "model path(" << (prop->model_files[0] ? prop->model_files[0] : "<NULL>");
+  std::cout << ") is not used. The model path is hardcoded within the API." << std::endl;
+}
+
+/**
+ * @brief Invoke flare using input tensors
+ */
+void
+TensorFilterFlare::invoke (const GstTensorMemory *input, GstTensorMemory *output)
+{
+  UNUSED (input);
+  UNUSED (output);
+  throw std::runtime_error ("Flare only supports invoke-dynamic mode. Set `invoke-dynamic=true`");
+}
+
+/**
+ * @brief Invoke flare using input tensors
+ */
+void
+TensorFilterFlare::invoke_dynamic (GstTensorFilterProperties *prop,
+    const GstTensorMemory *input, GstTensorMemory *output)
+{
+  std::string prompt;
+  std::string output_text;
+  SFlareApi::ErrorCode result;
+
+  if (!input || !input[0].data || input[0].size == 0) {
+    throw std::invalid_argument ("Invalid input tensor data");
+  }
+
+  prompt.assign ((char *) input[0].data, input[0].size);
+
+  result = context->executeSFlareLLM (prompt, output_text, output_size);
+  if (result != SFlareApi::ErrorCode::SFLARE_SUCCESS) {
+    throw std::runtime_error ("Failed to execute sflare LLM");
+  }
+
+  /* Stores string length, excluding NULL appended to the end of the string */
+  output[0].size = output_text.length ();
+  output[0].data = strndup (output_text.c_str (), output_text.length ());
+
+  gst_tensors_info_free (&prop->output_meta);
+  prop->output_meta.num_tensors = 1;
+  prop->output_meta.format = _NNS_TENSOR_FORMAT_FLEXIBLE;
+  prop->output_meta.info[0].type = _NNS_UINT8;
+  prop->output_meta.info[0].dimension[0] = output[0].size;
+}
+
+/**
+ * @brief Get framework info.
+ */
+void
+TensorFilterFlare::getFrameworkInfo (GstTensorFilterFrameworkInfo &info)
+{
+  info.name = name;
+  info.allow_in_place = 0;
+  info.allocate_in_invoke = 1;
+  info.run_without_model = 0;
+  info.verify_model_path = 1;
```

Comment:
prop->model_files in configure_instance()

If flare does not need model file (user does not set a model path in pipeline), set 'verify_model_path' as 0 to avoid checking model path in tensor-filter.

Reply:
Currently, the model file is hardcoded inside the Flare API. This will be fixed.
Thank you.